src/mame/drivers/blit.cpp

```diff
@@ -0,0 +1,297 @@
+	// octal 0070 - 16-bit
+	map(0x060038, 0x060039).w(FUNC(blit_state::vblank_ack));
+}
+
+static INPUT_PORTS_START(blit)
+INPUT_PORTS_END
+
+static DEVICE_INPUT_DEFAULTS_START( host_rs232_defaults )
+	DEVICE_INPUT_DEFAULTS( "RS232_TXBAUD", 0xff, RS232_BAUD_19200 )
+	DEVICE_INPUT_DEFAULTS( "RS232_RXBAUD", 0xff, RS232_BAUD_19200 )
+	DEVICE_INPUT_DEFAULTS( "RS232_STARTBITS", 0xff, RS232_STARTBITS_1 )
+	DEVICE_INPUT_DEFAULTS( "RS232_DATABITS", 0xff, RS232_DATABITS_8 )
+	DEVICE_INPUT_DEFAULTS( "RS232_PARITY", 0xff, RS232_PARITY_NONE )
+	DEVICE_INPUT_DEFAULTS( "RS232_STOPBITS", 0xff, RS232_STOPBITS_2 )
+	DEVICE_INPUT_DEFAULTS( "FLOW_CONTROL", 0x01, 0x01 )
+DEVICE_INPUT_DEFAULTS_END
+
+static DEVICE_INPUT_DEFAULTS_START( kbd_rs232_defaults )
+	DEVICE_INPUT_DEFAULTS( "RS232_TXBAUD", 0xff, RS232_BAUD_19200 )
+	DEVICE_INPUT_DEFAULTS( "RS232_RXBAUD", 0xff, RS232_BAUD_19200 )
+	DEVICE_INPUT_DEFAULTS( "RS232_STARTBITS", 0xff, RS232_STARTBITS_1 )
+	DEVICE_INPUT_DEFAULTS( "RS232_DATABITS", 0xff, RS232_DATABITS_8 )
+	DEVICE_INPUT_DEFAULTS( "RS232_PARITY", 0xff, RS232_PARITY_NONE )
+	DEVICE_INPUT_DEFAULTS( "RS232_STOPBITS", 0xff, RS232_STOPBITS_2 )
+	DEVICE_INPUT_DEFAULTS( "FLOW_CONTROL", 0x01, 0x01 )
+DEVICE_INPUT_DEFAULTS_END
+
+
+void blit_state::start_write(offs_t offset, uint16_t data, uint16_t mem_mask)
+{
+	LOG("START <- %04X (addr %06X)\n", data, data << 2);
+	m_videostart = data << 1;
+}
+
+WRITE_LINE_MEMBER(blit_state::system_clock_write)
+{
+	m_acia0->write_txc(state);
+	m_acia0->write_rxc(state);
+	//
+	m_acia1->write_txc(state);
+	m_acia1->write_rxc(state);
+}
+
+void blit_state::vblank_ack(offs_t offset, uint16_t data, uint16_t mem_mask)
+{
+	m_maincpu->set_input_line(M68K_IRQ_1, CLEAR_LINE);
+}
+
+
+uint32_t blit_state::screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect)
+{
+	int fg, bg;
+
+	if (BIT(*m_misccr, 0))
+	{
+		fg = 0; bg = 1;
+	}
+	else
+	{
+		fg = 1; bg = 0;
+	}
+
+	for (int y = 0; y < 1024; y++)
+	{
+		uint16_t *p = &bitmap.pix(y);
+
+		for (int x = 0; x < 800 / 16; x += 1)
+		{
+			uint16_t gfx;
+
+			gfx = m_p_ram[m_videostart + x + y * 50];
+
+			for (int z = 15; z >= 0; z--)
+			{
+				*p++ = BIT(gfx, z) ? fg : bg;
+			}
+		}
+	}
+	return 0;
+}
+
+
+void blit_state::machine_start()
+{
+}
+
+void blit_state::machine_reset()
+{
+	address_space &program = m_maincpu->space(AS_PROGRAM);
+	program.install_rom(0x000000, 0x000007, m_sysrom);   // do it here for F3
+	m_rom_shadow_tap = program.install_read_tap(0x040000, 0x045fff, "rom_shadow_r",[this](offs_t offset, u16 &data, u16 mem_mask)
+	{
+		if (!machine().side_effects_disabled())
+		{
+			// delete this tap
+			m_rom_shadow_tap->remove();
+
+			// reinstall ram over the rom shadow
+			m_maincpu->space(AS_PROGRAM).install_ram(0x000000, 0x000007, m_p_ram);
+		}
+
+		// return the original data
+		return data;
+	});
+
+	*m_misccr = 0;
+
+	m_maincpu->reset();
```

Comment:
This is also not necessary, because since 8f7757297aa5115c3d1d97bad8af55e231caef79 M68000 CPUs no longer perform the initial SP and PC fetches at `device_reset` time.

Reply:
ok - removed.